python/argo/connection.py

```diff
@@ -73,15 +78,23 @@ def setup(self) -> None:
             if match:
                 self.port = int(match.group(1))
             else:
-                raise Exception("Failed to load process, output was `" + \
+                raise Exception("Failed to load process, output was `" +
                                 out_line + "' but expected PORT then a port.")
 
+    def pid(self) -> Optional[int]:
+        """Return the process group id of the managed server process"""
+        if self.proc is not None:
+            return os.getpgid(self.proc.pid)
+        else:
+            return None
+
     def __del__(self) -> None:
         if self.proc is not None:
-            try:
-                os.killpg(os.getpgid(self.proc.pid), signal.SIGKILL)
-            except ProcessLookupError:
-                pass
+            if not self.persist:
```

Comment:
This is the important bit: when the process handle is GC'ed, we don't kill the underlying process if `self.persist` is set to True.